Cargo.toml

```diff
@@ -55,7 +55,7 @@ resolver = "2"
 
 [workspace.package]
 name = "fedimint"
-version = "0.3.1-rc.1"
+version = "0.3.2-rc.0"
```

Comment:
We really should `= { workspace = true}` everything here. Then this PR would be a one liner.

Reply:
BTW. Why are we updating from `0.3.1-rc.1` instead of `0.3.1`?

Reply:
I branched off from the `v0.3.1-rc.1` tag to the `v0.3.1` branch since there were already more changes on top and we didn't want to go through another round of testing.

![Untitled Diagram drawio(5)](https://github.com/fedimint/fedimint/assets/84478054/31ff7087-7856-4679-bd5d-69074882f073)

Reply:
> We really should `= { workspace = true}` everything here. Then this PR would be a one-liner.

Let's do that on master (at least some crates were already switched to that scheme) so 0.4 will have it. These changes are 100% automated, so not a huge deal imo.